### PR TITLE
create academic goals rollup

### PR DESCRIPTION
**Pull request by anthonygwalters**
# Pull Request

## Summary & Motivation

[//]: # "When merged, this pull request will..."

## Self-review

- [ ] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [ ] <kbd>Format</kbd> has been run on all modified files

- [ ] Ensure you are using the `union_dataset_join_clause()` macro for queries that employ any
      models from these datasets:
  - deanslist
  - edplan
  - iready
  - pearson
  - powerschool
  - renlearn
  - titan

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

### CODE REVIEW

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/iready/base/base_iready__diagnostic_results.sql [hunk 1]
```diff
@@ -147,39 +171,46 @@ from diagnostic_results as dr
...
+    on dr.subject = cws.test_name
     and dr.student_grade = cws.grade_level_string
     and dr.state_assessment_type = cws.destination_system
+    and dr.overall_scale_score_plus_stretch_growth
+    between cws.scale_low and cws.scale_high
     and cws.source_system = 'i-Ready'
+left join
+    {{ ref("stg_assessments__iready_crosswalk") }} as cwi
+    on dr.subject = cwi.test_name
+    and dr.student_grade_int = cwi.grade_level
```
[the remark this answers is not in the document]
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/iready/base/base_iready__diagnostic_results.sql:210]: K has never been needed before, and this is the only join where K exists

File: src/dbt/kipptaf/models/reporting/intermediate/int_reporting__student_filters.sql [hunk 2]
```diff
@@ -75,9 +56,33 @@ with
...
+                pt.score_type = 'sat10_eb_read_write_section_score', 'ELA', 'Math'
+            ) as discipline,
+            if(max(pt.score) >= 420, 'Bucket 1', null) as bucket_1,
+        from {{ ref("int_illuminate__psat_unpivot") }} as pt
+        inner join
+            {{ ref("base_powerschool__student_enrollments") }} as co
+            on safe_cast(pt.local_student_id as int) = co.student_number
+            and pt.academic_year = co.academic_year
+            and co.rn_year = 1
+            and co.grade_level = 10
```
[the remark this answers is not in the document]
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/reporting/intermediate/int_reporting__student_filters.sql:85]: to get only tests that were taken when the kid was in 10th grade

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__academic_goals_rollup.sql [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+
+    grade_bands as (
+        select 'K' as band, 0 as grade_level,
+        union all
+        select '1-2' as band, grade_level,
+        from unnest([1, 2]) as grade_level
+        union all
+        select '3-8' as band, grade_level,
+        from unnest([3, 4, 5, 6, 7, 8]) as grade_level
+    ),
```
[the remark this answers is not in the document]
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__academic_goals_rollup.sql:20]: the K split out is a little ad hoc & I don't anticipate it being needed elsewhere, so probably not? I defer to you.

File: src/dbt/kipptaf/models/iready/base/base_iready__diagnostic_results.sql [hunk 4]
```diff
@@ -147,39 +171,46 @@ from diagnostic_results as dr
...
-    on dr.overall_scale_score_plus_stretch_growth
-    between cws.scale_low and cws.scale_high
-    and dr.subject = cws.test_name
+    on dr.subject = cws.test_name
     and dr.student_grade = cws.grade_level_string
     and dr.state_assessment_type = cws.destination_system
+    and dr.overall_scale_score_plus_stretch_growth
+    between cws.scale_low and cws.scale_high
     and cws.source_system = 'i-Ready'
+left join
```
[the remark this answers is not in the document]
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/iready/base/base_iready__diagnostic_results.sql:207]: no, that column is null for these values

**Review by cbini — APPROVED** [review]

**Pull request merged**